doc: local executors - rephrase scheduler process (#32983)
mistercrunch/Airflow: docs/apache-airflow/core-concepts/executor/index.rst

.. note::

>     Something that often confuses new users of Airflow is that they don't need to run a separate ``executor`` process using one of the Local Executors or Remote Executors (because executor's logic runs *inside* the ``scheduler`` process and this logic will run the tasks locally or not depending the executor selected).

Reviewer: ```suggestion
    New Airflow users may assume they need to run a separate executor process using one of the Local or Remote Executors. This is not correct. The executor logic runs *inside* the scheduler process, and will run the tasks locally or not depending the executor selected.
```

I needed to read the first part of the sentence several times to understand. Flipping the conditional may be useful…

Author: thanks for the review
